stake-pool/cli/src/main.rs

```diff
@@ -1283,8 +1283,9 @@ fn prepare_withdraw_accounts(
             continue;
         }
 
-        let available_for_withdrawal_wo_fee =
-            stake_pool.calc_pool_tokens_for_deposit(lamports).unwrap();
+        let available_for_withdrawal_wo_fee = stake_pool
+            .calc_pool_tokens_for_deposit(lamports - min_balance)
+            .unwrap();
```

Comment:
Now that I look through the code, this should only be an issue with transient stake accounts and the reserve. `transient_stake_lamports` includes the rent exemption + minimum, but `active_stake_lamports` doesn't.  Can you instead change line 1260 to read `validator.transient_stake_lamports - min_balance`? And then line 1268 should be ` reserve_stake.lamports - rpc_client
.get_minimum_balance_for_rent_exemption(STAKE_STATE_LEN)? - 1`

But another bit first -- has this been observed for the validator stake accounts too?  If so, we have a potentially larger bug to look at.

Reply:
is there any difference between active stake accounts and transient stake accounts? both of them have  rent_exemption and MINIMUM_ACTIVE_STAKE ?
@CMEONE had that issue - https://github.com/solana-labs/solana-program-library/pull/2806#issuecomment-1033323474 , not sure if that were transient or active stake accounts

Reply:
> is there any difference between active stake accounts and transient stake accounts?

Yes, the lamports reported are *not* the same. `transient_stake_lamports` includes the rent exemption + minimum, but `active_stake_lamports` doesn't.  So for example, if the stake pools says `active_stake_lamports` == 10 SOL, the validator stake account actually has 10.0032.... SOL. But if it says `transient_stake_lamports` == 10 SOL, the transient stake account has exactly 10 SOL.

> had that issue - https://github.com/solana-labs/solana-program-library/pull/2806#issuecomment-1033323474 , not sure if that were transient or active stake accounts

I have a feeling they were trying to withdraw from a transient account.

Reply:
The stake accounts that I tried to withdraw were transient stake accounts. I moved funds from reserve into validator stake accounts and tried withdrawing stake accounts (transient because they were still activating) from the pool within the same epoch.